src/container/Header/Header.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState } from 'react';
+
+import { ProductData, UserData } from 'mockData';
+import { Link, useNavigate } from 'react-router';
+import { getFormattedRoute } from 'utils';
+
+import {
+    Menu as MenuIcon,
+    Notifications as NotificationsIcon,
+} from '@mui/icons-material';
+import {
+    Avatar,
+    Box,
+    IconButton as MuiIconButton,
+    Typography,
+} from '@mui/material';
+
+import { AutoComplete, IconButton, Popover } from '@component';
+import { Images, routes } from '@constant';
+
+import { HeaderBox, StyledHeader } from './Header.styles';
+import { HeaderProps } from './Header.types';
+
+export const Header = ({
+    showHamburger,
+    onMenuClick,
+    ...rest
+}: HeaderProps) => {
+    const navigate = useNavigate();
+    const [options, setOptions] = useState<string[]>([]);
+
+    useEffect(() => {
+        const productTitles = ProductData.map((item) => item.title);
+        setOptions(productTitles);
+    }, []);
+
+    const handleChange = (
+        _: React.SyntheticEvent<Element, Event>,
+        value: string,
+    ) => {
+        if (typeof value === 'string' && value.trim() !== '') {
+            const slug = encodeURIComponent(value.trim().toLowerCase());
+            const productRoute = getFormattedRoute(routes.Product, [slug]);
+            navigate(productRoute);
+        }
+    };
+
+    return (
+        <StyledHeader {...rest}>
+            {!showHamburger ? (
+                <HeaderBox gap={8}>
+                    <Link to={routes.Home} style={{ fontSize: 0 }}>
+                        <img src={Images.HeaderLogo} alt="Logo" />
+                    </Link>
+                    <AutoComplete
+                        options={options}
+                        placeholder="Search..."
+                        size="small"
+                        onChange={handleChange}
+                    />
+                </HeaderBox>
+            ) : (
+                <IconButton icon={MenuIcon} onClick={onMenuClick} />
+            )}
+            <HeaderBox gap={3}>
+                <IconButton
+                    icon={NotificationsIcon}
+                    rounded
+                    color="secondary"
+                    isBoxShadow
+                />
+                <Popover
+                    trigger={({ onClick }) => (
+                        <MuiIconButton onClick={onClick}>
+                            <Avatar src={UserData.imageUrl} />
+                        </MuiIconButton>
+                    )}
+                >
+                    <Box padding={3}>
+                        <Typography variant="h3">{UserData.name}</Typography>
```

Comment:
```suggestion
                        <Typography variant="h3">{USER_DATA.name}</Typography>
```
Explore naming conventions for variables.
https://www.uxpin.com/studio/blog/react-best-practices/